spec/rails/time_difference_spec.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Timely::TimeDifference do
+  def self.with_each_class(&block)
+    classes = [Time, Date, DateTime]
+
+    classes.each do |clazz|
+      context "with a #{clazz.name} class" do
+        instance_exec clazz, &block
+      end
+    end
+  end
+
+  describe '.between' do
+    with_each_class do |clazz|
+      it 'returns a new TimeDifference instance in each component' do
+        start_time = clazz.new(2011, 1)
```

Comment:
You could cut out a lot of duplication if you moved `start_time` and `end_time` into let variables at the top

Reply:
As discussed this is a port of the upstream library I didn't want to spend any time refactoring it